tests/internal/test_process_graph_visitor.py

```diff
@@ -126,21 +126,21 @@ def test_visit_nodes_array():
 def test_visit_array_with_dereferenced_nodes():
     graph = {
         'arrayelement1': {
-            'arguments': {'data': {'from_argument': 'data'}, 'index': 2},
```

Comment:
I'd leave this `test_visit_array_with_dereferenced_nodes` untouched and make a copy to test parameter handling specifically